src/network/ip_address.cpp

```diff
@@ -40,6 +40,10 @@ std::array<uint8_t, 4> parse(const std::string& ip)
     int b = -1;
     int c = -1;
     int d = -1;
+
+    if (ip.empty())
+        return {{static_cast<uint8_t>(a), static_cast<uint8_t>(b), static_cast<uint8_t>(c), static_cast<uint8_t>(d)}};
+
```

Comment:
The caller should create an IPAddress object with 0xFFFFFFFF instead of giving an empty string.